drivers/adc/ad4170/ad4170.c

```diff
@@ -0,0 +1,2123 @@
+	else {
+		_memswap64(&data, reg_size, reg_size);
+		memcpy(&buf[i], &data, reg_size);
+	}
+	i += reg_size;
+
+	sz = i;
+
+	if (dev->spi_settings.crc_enabled) {
+		ocrc = no_os_crc8(ad4170_crc8, buf, i, AD4170_CRC8_INITIAL_VALUE);
+		buf[i++] = ocrc;
+		sz = i;
+	}
+
+	ret = no_os_spi_write_and_read(dev->spi_desc, buf, sz);
+	if (ret < 0)
+		return ret;
+
+	ret = _verify_sync_pattern(dev, buf);
+	if (ret < 0)
+		return ret;
+
+	if (dev->spi_settings.crc_enabled) {
+		if (ocrc != buf[sz - 1])
+			return -EBADMSG;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief SPI write device register using a mask.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param mask - The mask.
+ * @param reg_data - The register data.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad4170_spi_reg_write_mask(struct ad4170_dev *dev,
+				  uint32_t reg_addr,
+				  uint8_t mask,
+				  uint32_t reg_data)
+{
+	uint32_t reg;
+	int32_t ret;
+
+	ret = ad4170_spi_reg_read(dev, reg_addr, &reg);
+	if (ret < 0)
+		return ret;
+
+	reg &= ~mask;
+	reg |= reg_data;
+
+	return ad4170_spi_reg_write(dev, reg_addr, reg);
+}
+
+/**
+ * @brief Reset the SPI interface by sending reset sequence to device
+ * @param dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad4170_reset_spi_interface(struct ad4170_dev *dev)
+{
+	int32_t ret;
+
+	/* Data sequence to reset AD4170 SPI interface (71*1 + 1*0) * 3
+	 * Note: CS must be low for entire sequence. */
+	uint8_t buf[] = {0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xfe,
+			 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xfe,
+			 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xfe
+			};
+
+	if (!dev)
+		return -EINVAL;
+
+	ret = no_os_spi_write_and_read(dev->spi_desc, buf, sizeof(buf));
+	if (ret < 0)
+		return ret;
+
+	return 0;
+}
+
+/**
+ * @brief Get data from register Data_16b.
+ * @param dev - The device structure.
+ * @param data - Output pointer where to store the data value.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad4170_get_data16(struct ad4170_dev *dev, uint16_t *data)
+{
+	uint32_t reg;
+	int32_t ret;
+
+	if (!data)
+		return -EINVAL;
+
+	ret = ad4170_spi_reg_read(dev, AD4170_REG_DATA_16b, &reg);
+	if (ret < 0)
+		return ret;
+
+	*data = (uint16_t)reg;
+
+	return 0;
+}
+
+/**
+ * @brief Get data and status from register Data_16b_Status.
+ * @param dev - The device structure.
+ * @param data - Output pointer where to store the data value.
+ * @param status - Output pointer where to store the status value.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad4170_get_data16s(struct ad4170_dev *dev, uint16_t *data,
+			   uint8_t *status)
+{
+	uint32_t reg;
+	int32_t ret;
+
+	if (!data || !status)
+		return -EINVAL;
+
+	ret = ad4170_spi_reg_read(dev, AD4170_REG_DATA_16b_STATUS, &reg);
+	if (ret < 0)
+		return ret;
+
+	*status = (uint8_t)reg;
```

Comment:
In cases like these, I think we usually use `no_os_field_get()` along with some macros for the bit mask of the register field. I'm fine with either option.